### PR TITLE
Add download_pvolfiles

### PR DESCRIPTION
**Pull request by Rafnuss**
Addition of a function to download pvol file from AWS 

Some keys decision I made:
- Code follow  `download_vpfiles.R` as closely as possible
- Rely on the package [`aws.s3`](https://github.com/cloudyr/aws.s3). I've added it to `DESCRIPTION` as suggested. Note sure if that's correct usage. 
- The function does not accept multiple radar code. 
- `date_min` and `date_max` have to be set with hours and minutes.
- I've added a progess bar `setTxtProgressBar` from the `utils` pacakge. do you need `utils::setTxtProgressBar` ?. [`r-lib/progress`](https://github.com/r-lib/progress) would probably be nicer, but require a new package. What do you think?
- I wrote the test page that goes with it.

Related issues:  #41

### CODE REVIEW

**Comment by Rafnuss** [comment]
Hi @adokter , there were indeed a few lines left from the download_vpts file. I am not sure how this happened. All checks have passed now. Hope this is good to go

**Comment by adokter** [comment]
Thanks @Rafnuss, thanks almost ready, just a few small things, see above review

**Review by adokter — CHANGES_REQUESTED** [review]

**Comment by bart1** [comment]
> 
> Yes, good idea. I was just trying to follow [`download_vpfiles.R`](https://github.com/adokter/bioRad/blob/master/R/download_vpfiles.R). Should we change both at the same time then ?
> 
>     * Agree that standard date format should be accepted. I still haven't figure out what is the standard in R, seems rather unclear.

`POSIXct` is pretty much the de facto standard in R for timestamps, the `date` format for dates. `POSIXct` internally is as unix timestamp stored as a double. As a general comment for handling timestamps the `lubridate` package is really nice. It for example has an `%within%` function to test if a date is within an interval
 
>     * I am not convinced of the full day option. It's relatively easy for the user to specify both date. But if there is a consensus toward that standard, I can implement it.

It would make some operations easier like parallel operations e.g.:
```R
parallel::mclapply(as.Date("2016-1-1")+1:10, download_pvolfiles, radar='xxx')
```
Its not really crucial but I thought it might be an easy add

> 
>     * Often we only want the night data. In my Matlab code, I have a parameter that check this. Is there something easy/already exisiting in bioRad to do that?

The `bioRad::sunset` function identifies the right times. If the function would accept lubridate intervals something like this could work
```R
ints<-lubridate::interval(sunset(Sys.Date()+1:10,1,1), sunrise(Sys.Date()+2:11,1,1))
lapply(ints, download_pvolfiles, radar='xxx')
# or if a vector if ints could be fed this could work
download_pvolfiles(ints, radar='xxx')
```
> 
>     * I'll check about timezone. The data on aws is stored in UTC, so I guess we could have at timezone parameter. But coming back to your first point, I would almost be in favor to only accept standard date type which have timezone incorporated in them by default. Then we don't have to worry about that.
> 
`POSIXct` incorporates a timestamp and for that reason might be a good choice.

> 
> Let me know what you think and I can give it a go.

File: R/download_pvolfiles.R [hunk 1]
```diff
@@ -23,22 +22,45 @@
...
-  assert_that(radar %in% possible_radar, msg = paste0("radar ", radar, " doesn't exist"))
+  assert_that(length(radar) == 1, msg = paste0("radar is not of length 1"))
+  possible_radar <- c("KABR", "KABX", "KAKQ", "KAMA", "KAMX", "KAPX", "KARX",
```
[the remark this answers is not in the document]
Comment (Rafnuss) [reply R/download_pvolfiles.R]: Ok, suggesting to do that with a `trycatch`: 
https://github.com/adokter/bioRad/blob/06082eb39e4c271c965c50d16a4573a03f212c91/R/download_pvolfiles.R#L76-L85

File: R/download_pvolfiles.R [hunk 2]
```diff
@@ -49,53 +71,48 @@ download_pvolfiles <- function(date_min, date_max, radar, directory = ".",
   check_date_format(date_max, "%Y-%m-%d %H:%M")
 
   # Stop if overwrite is not a logical
-  assert_that(is.logical(overwrite), msg='overwrite is not a logical')
+  assert_that(is.logical(overwrite), msg = "overwrite is not a logical")
 
   # Create series of unique days yyyy-mm-dd based on date_min/max:
   dates <- seq(
     as.Date(substring(date_min, 1, 10), tz = NULL),
```
[the remark this answers is not in the document]
Comment (Rafnuss) [reply R/download_pvolfiles.R]: OK, https://github.com/adokter/bioRad/blob/06082eb39e4c271c965c50d16a4573a03f212c91/R/download_pvolfiles.R#L45-L46

File: R/download_pvolfiles.R [hunk 3]
```diff
@@ -49,53 +71,48 @@ download_pvolfiles <- function(date_min, date_max, radar, directory = ".",
...
+        dd <= strptime(date_max, format = "%Y-%m-%d %H:%M")
     })
-    bucket_df = bucket_df[isWithin, ]
+    bucket_df <- bucket_df[isWithin, ]
 
     # create progresbar
-    message(paste0("Downloading pvol for ",prefix))
-    pb = txtProgressBar(min = 0, max = nrow(bucket_df), initial = 0, style = 3)
+    message(paste0("Downloading pvol for ", prefix))
+    pb <- txtProgressBar(min = 0, max = nrow(bucket_df), initial = 0, style = 3)
```
[the remark this answers is not in the document]
Comment (Rafnuss) [reply R/download_pvolfiles.R:124]: https://github.com/adokter/bioRad/blob/06082eb39e4c271c965c50d16a4573a03f212c91/R/download_pvolfiles.R#L123

**Comment by Rafnuss** [comment]
Ok, I have done most of the suggestions with the exception of :
- Accepting full day download or date format (only`POSIXct`)
- Night/Day filtering
- parallelisation

**Comment by peterdesmet** [comment]
Thanks for this @Rafnuss. Note that I will likely update `download_vpfiles.R` in the future, to download more convenient tabular files (https://github.com/enram/vpts-dp/issues/25) from the repository rather than the raw vp hdf5 files. I will then assess the start/end parameters too (and likely use posix).

**Pull request merged**